File: Year 3/Fall Semester/CS-315_Programming_Languages/Homework1/Codes/gultekin_omeroktay.lua
function foo(key, value)
    print(key, value)
end

function printDict(dict)
  print()
  for k, v in pairs(dict) do
    foo(k, v)
  end
  print()
end

function hasValue(dict, val)
    for k, v in pairs(dict) do
      if v == val then
            return true
      end
    end
    return false
end

print("--------------------------- 1 ---------------------------");
cs315_instructors = {
    ["Section 1"] = "Altay Güvenir",
    ["Section 2"] = "Aynur Dayanik",
    ["Section 3"] = "Karani Kardas",
    Section       = "Different way to create key"
}

print(cs315_instructors)
printDict(cs315_instructors)
print("--------------------------- 1 ---------------------------");

print("--------------------------- 2 ---------------------------");
print(cs315_instructors["Section 1"])
print("--------------------------- 2 ---------------------------");

print("--------------------------- 3 ---------------------------");
cs315_instructors["Section 4"] = "Not Available"

printDict(cs315_instructors)
print("--------------------------- 3 ---------------------------");

print("--------------------------- 4 ---------------------------");
cs315_instructors["Section 4"] = nil

printDict(cs315_instructors)
print("--------------------------- 4 ---------------------------");

print("--------------------------- 5 ---------------------------");
cs315_instructors["Section 1"] = "Halil Altay Güvenir"

printDict(cs315_instructors)
print("--------------------------- 5 ---------------------------");

print("--------------------------- 6 ---------------------------");
--[[
~= is negation operator in lua
--]]

print(cs315_instructors["Section 1"] ~= nil)
print("--------------------------- 6 ---------------------------");

print("--------------------------- 7 ---------------------------");
print(hasValue(cs315_instructors, "Karani Kardas"));
print("--------------------------- 7 ---------------------------");

print("--------------------------- 8 ---------------------------");
print()
for k, v in pairs(cs315_instructors) do
    foo(k, v)
end
print()

print("---------------------------  8 ---------------------------");

File: Year 3/Fall Semester/CS-315_Programming_Languages/Homework1/Codes/gultekin_omeroktay.rb
def foo(k, v)
  puts format('%s : %s', k, v)
end

print "--------------------------- 1 ---------------------------", "\n"
cs315_instructors = {
  'Section 1' => 'Altay Güvenir',
  'Section 2' => 'Aynur Dayanik',
  'Section 3' => 'Karani Kardas'
}

print cs315_instructors, "\n"
print "--------------------------- 1 ---------------------------", "\n"

print "--------------------------- 2 ---------------------------", "\n"
print cs315_instructors['Section 3'], "\n"
print "--------------------------- 2 ---------------------------", "\n"

print "--------------------------- 3 ---------------------------", "\n"
cs315_instructors['Section 4'] = 'Not Available'
print cs315_instructors, "\n"
print "--------------------------- 3 ---------------------------", "\n"

print "--------------------------- 4 ---------------------------", "\n"
puts cs315_instructors.delete('Section 4')
print "#{cs315_instructors}\n"
print "--------------------------- 4 ---------------------------", "\n"

print "--------------------------- 5 ---------------------------", "\n"
cs315_instructors['Section 1'] = 'Halil Altay Güvenir'
print "#{cs315_instructors}\n"
print "--------------------------- 5 ---------------------------", "\n"

print "--------------------------- 6 ---------------------------", "\n"
print cs315_instructors.has_key?('Section 1'), "\n"
print "--------------------------- 6 ---------------------------", "\n"

print "--------------------------- 7 ---------------------------", "\n"
print cs315_instructors.has_value?('Karani Kardas'), "\n"
print "--------------------------- 7 ---------------------------", "\n"

print "--------------------------- 8 ---------------------------", "\n"
cs315_instructors.each_pair { |k, v| foo(k, v) }
print "--------------------------- 8 ---------------------------", "\n"
